blockchain/blockchain.go

```diff
@@ -209,3 +233,47 @@ func (b *Blockchain) SanityCheckNewHeight(block *core.Block, stateUpdate *core.S
 
 	return core.VerifyBlockHash(block, b.network)
 }
+
+func (b *Blockchain) StoreL1ChainHeight(height uint64, block *core.Block) error {
+	if err := b.VerifyL1ChainHeight(height); err != nil {
```

Comment:
this shouldn't be creating a new txn. We should drop the call to `VerifyL1ChainHeight` and call `verifyL1ChainHeight` inside the Update() callback